core/src/cluster_nodes.rs

```diff
@@ -462,22 +479,60 @@ impl From<Pubkey> for NodeId {
     }
 }
 
+// pub for bench
+pub fn make_cluster<R: Rng>(
+    rng: &mut R,
+    num_nodes: usize,
+) -> (
+    Vec<ContactInfo>,
+    HashMap<Pubkey, u64>, // stakes
+    ClusterInfo,
+) {
+    let mut nodes: Vec<_> = repeat_with(|| ContactInfo::new_rand(rng, None))
+        .take(num_nodes)
+        .collect();
+    nodes.shuffle(rng);
+    let this_node = nodes[0].clone();
+    let mut stakes: HashMap<Pubkey, u64> = nodes
+        .iter()
+        .filter_map(|node| {
+            if rng.gen_ratio(1, 7) {
+                None // No stake for some of the nodes.
+            } else {
+                Some((node.id, rng.gen_range(0, 20)))
+            }
```

Comment:
Out of curiosity, how does this ratio of `unstaked:staked` nodes of `1:6` affect the performance of the shuffle? Does more or less unstaked/staked nodes make perf faster?

Reply:
Added some tests for comparison:

```
running 4 tests
test bench_get_retransmit_peers_compat_unstaked_ratio_1_2         ... bench:   1,679,677 ns/iter (+/- 2,628)
test bench_get_retransmit_peers_compat_unstaked_ratio_1_32        ... bench:   1,690,616 ns/iter (+/- 3,048)
test bench_get_retransmit_peers_deterministic_unstaked_ratio_1_2  ... bench:   1,478,714 ns/iter (+/- 13,556)
test bench_get_retransmit_peers_deterministic_unstaked_ratio_1_32 ... bench:   2,476,904 ns/iter (+/- 39,032)

test result: ok. 0 passed; 0 failed; 0 ignored; 4 measured; 0 filtered out; finished in 5.22s
```

Reply:
thanks for the insights!